Couple of fuzzer fail when parallel execution information is passed down

With https://github.com/SymbiFlow/prjxray/pull/596 the parallel make information is successfully passed down to submake files and it seems that the `100-dsp-mskpat` fuzzer seems to get confused in some way;
```
100-dsp-mskpat           - 3m04s: Finishing (User:772s System:27s).
100-dsp-mskpat           - 3m04s: Failed @ 2019-02-05T02:41:47.869730 with exit code: 2
100-dsp-mskpat           - 3m04s: --------------------------------------------------------------------------
100-dsp-mskpat           - 3m04s: ++ cd build/specimen_008
100-dsp-mskpat           - 3m04s: ++ cd build/specimen_001
100-dsp-mskpat           - 3m04s: ++ export SEED=7ee7ec67
100-dsp-mskpat           - 3m04s: ++ export SEED=fad8c932
100-dsp-mskpat           - 3m04s: ++ SEED=7ee7ec67
100-dsp-mskpat           - 3m04s: ++ SEED=fad8c932
100-dsp-mskpat           - 3m04s: +++ echo build/specimen_001
100-dsp-mskpat           - 3m04s: +++ md5sum
100-dsp-mskpat           - 3m04s: +++ md5sum
100-dsp-mskpat           - 3m04s: +++ cut -c1-8
100-dsp-mskpat           - 3m04s: +++ cut -c1-8
100-dsp-mskpat           - 3m04s: ++ export SEEDN=6
100-dsp-mskpat           - 3m04s: ++ SEEDN=6
100-dsp-mskpat           - 3m04s: ++ export SEEDN=4
100-dsp-mskpat           - 3m04s: ++ SEEDN=4
100-dsp-mskpat           - 3m04s: +++ sed 's/specimen_0*//'
100-dsp-mskpat           - 3m04s: ++++ pwd
100-dsp-mskpat           - 3m04s: ++++ pwd
100-dsp-mskpat           - 3m04s: +++ cut -c1-8
100-dsp-mskpat           - 3m04s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/100-dsp-mskpat/build/specimen_007
100-dsp-mskpat           - 3m04s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/100-dsp-mskpat/build/specimen_002
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_006
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_004
100-dsp-mskpat           - 3m04s: +++ sed 's/specimen_0*//'
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: ++ export SEED=ddcf076c
100-dsp-mskpat           - 3m04s: ++ SEED=ddcf076c
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: ++ export SEED=22b443a6
100-dsp-mskpat           - 3m04s: ++ SEED=22b443a6
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: +++ sed 's/specimen_0*//'
100-dsp-mskpat           - 3m04s: ++ export SEEDN=7
100-dsp-mskpat           - 3m04s: ++ SEEDN=7
100-dsp-mskpat           - 3m04s: ++++ pwd
100-dsp-mskpat           - 3m04s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/100-dsp-mskpat/build/specimen_001
100-dsp-mskpat           - 3m04s: ++ export SEEDN=2
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''h27bbb5c2'
100-dsp-mskpat           - 3m04s: ++ SEEDN=2
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: +++ sed 's/specimen_0*//'
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_007
100-dsp-mskpat           - 3m04s: ++++ pwd
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_002
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''h152d8583'
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/100-dsp-mskpat/build/specimen_005
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: ++ export SEEDN=1
100-dsp-mskpat           - 3m04s: ++ SEEDN=1
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''h7ee7ec67'
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_001
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: ++ export SEEDN=5
100-dsp-mskpat           - 3m04s: ++ SEEDN=5
100-dsp-mskpat           - 3m04s: ++ cd build/specimen_003
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''hfad8c932'
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: +++ echo build/specimen_003
100-dsp-mskpat           - 3m04s: +++ md5sum
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_005
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''hddcf076c'
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: +++ cut -c1-8
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''h22b443a6'
100-dsp-mskpat           - 3m04s: ++ export SEED=131b315d
100-dsp-mskpat           - 3m04s: ++ SEED=131b315d
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: +++ sed 's/specimen_0*//'
100-dsp-mskpat           - 3m04s: ++++ pwd
100-dsp-mskpat           - 3m04s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/100-dsp-mskpat/build/specimen_003
100-dsp-mskpat           - 3m04s: ++ export SEEDN=3
100-dsp-mskpat           - 3m04s: ++ SEEDN=3
100-dsp-mskpat           - 3m04s: +++ echo build/specimen_008
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_003
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: ++ export SEED=258dce4a
100-dsp-mskpat           - 3m04s: ++ SEED=258dce4a
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''h131b315d'
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: +++ sed 's/specimen_0*//'
100-dsp-mskpat           - 3m04s: ++++ pwd
100-dsp-mskpat           - 3m04s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/100-dsp-mskpat/build/specimen_008
100-dsp-mskpat           - 3m04s: ++ export SEEDN=8
100-dsp-mskpat           - 3m04s: ++ SEEDN=8
100-dsp-mskpat           - 3m04s: ++ echo build/specimen_008
100-dsp-mskpat           - 3m04s: ++ md5sum
100-dsp-mskpat           - 3m04s: ++ cut -c1-8
100-dsp-mskpat           - 3m04s: + echo '`define SEED 32'\''h258dce4a'
100-dsp-mskpat           - 3m04s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
100-dsp-mskpat           - 3m04s: make[1]: *** [run] Error 2
100-dsp-mskpat           - 3m04s: 
100-dsp-mskpat           - 3m04s: --------------------------------------------------------------------------
100-dsp-mskpat           - 3m04s: Failed @ 2019-02-05T02:41:47.869730 with exit code: 2
Makefile:77: recipe for target '100-dsp-mskpat/run.ok' failed
make: *** [100-dsp-mskpat/run.ok] Error 2
make: *** Waiting for unfinished jobs....
```

Looks like it might affect other fuzzers too;
```
052-pip-clkin            - 1m40s: Still running (1m:91.97%, 5m:55.85%, 15m:96.92%).
052-pip-clkin            - 1m40s: make---sh---bash---make---20*[bash---make---vivado.sh---vivado---loader---vivado]
053-pip-ctrlin           - 1m40s: Still running (1m:91.97%, 5m:55.85%, 15m:96.92%).
053-pip-ctrlin           - 1m40s: make---sh---bash---make-+-10*[bash---make---sh---python3]
053-pip-ctrlin           - 1m40s:                         |-6*[bash---make---vivado.sh---vivado---loader---vivado]
053-pip-ctrlin           - 1m40s:                         `-bash---make
053-pip-ctrlin           - 1m50s: Finishing (User:2124s System:67s).
053-pip-ctrlin           - 1m50s: Failed @ 2019-02-05T14:12:38.089913 with exit code: 2
053-pip-ctrlin           - 1m50s: --------------------------------------------------------------------------
053-pip-ctrlin           - 1m50s: ++ export SEED=9aa8f41f
053-pip-ctrlin           - 1m50s: ++ SEED=9aa8f41f
053-pip-ctrlin           - 1m50s: +++ echo build/1/specimen_019
053-pip-ctrlin           - 1m50s: +++ cut -c1-8
053-pip-ctrlin           - 1m50s: +++ sed 's/specimen_0*//'
053-pip-ctrlin           - 1m50s: ++++ pwd
053-pip-ctrlin           - 1m50s: ++++ pwd
053-pip-ctrlin           - 1m50s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin/build/1/specimen_017
053-pip-ctrlin           - 1m50s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin/build/1/specimen_016
053-pip-ctrlin           - 1m50s: + export FUZDIR=/tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin
053-pip-ctrlin           - 1m50s: + FUZDIR=/tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin
053-pip-ctrlin           - 1m50s: + source /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/genheader.sh
053-pip-ctrlin           - 1m50s: ++ '[' -z artix7 ']'
053-pip-ctrlin           - 1m50s: ++ set -ex
053-pip-ctrlin           - 1m50s: ++ export FUZDIR=/tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin
053-pip-ctrlin           - 1m50s: ++ FUZDIR=/tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin
053-pip-ctrlin           - 1m50s: ++ test 1 -ge 1
053-pip-ctrlin           - 1m50s: ++ test '!' -e ''
053-pip-ctrlin           - 1m50s: ++ export SPECDIR=build/1/specimen_020
053-pip-ctrlin           - 1m50s: ++ SPECDIR=build/1/specimen_020
053-pip-ctrlin           - 1m50s: ++ mkdir -p build/1/specimen_020
053-pip-ctrlin           - 1m50s: +++ md5sum
053-pip-ctrlin           - 1m50s: ++ export SEED=c045d7c3
053-pip-ctrlin           - 1m50s: +++ cut -c1-8
053-pip-ctrlin           - 1m50s: ++ SEED=c045d7c3
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''h5bb070fb'
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: ++ export SEEDN=16
053-pip-ctrlin           - 1m50s: ++ SEEDN=16
053-pip-ctrlin           - 1m50s: ++ export SEEDN=15
053-pip-ctrlin           - 1m50s: ++ SEEDN=15
053-pip-ctrlin           - 1m50s: ++ echo build/1/specimen_016
053-pip-ctrlin           - 1m50s: ++ md5sum
053-pip-ctrlin           - 1m50s: ++ cut -c1-8
053-pip-ctrlin           - 1m50s: +++ sed 's/specimen_0*//'
053-pip-ctrlin           - 1m50s: +++ sed 's/specimen_0*//'
053-pip-ctrlin           - 1m50s: ++ echo build/1/specimen_015
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: ++++ pwd
053-pip-ctrlin           - 1m50s: ++ md5sum
053-pip-ctrlin           - 1m50s: ++ export SEED=9fd99230
053-pip-ctrlin           - 1m50s: ++ SEED=9fd99230
053-pip-ctrlin           - 1m50s: ++ cut -c1-8
053-pip-ctrlin           - 1m50s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin/build/1/specimen_018
053-pip-ctrlin           - 1m50s: ++ export SEEDN=17
053-pip-ctrlin           - 1m50s: ++ SEEDN=17
053-pip-ctrlin           - 1m50s: +++ sed 's/specimen_0*//'
053-pip-ctrlin           - 1m50s: ++ cd build/1/specimen_020
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''hd19e3576'
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: ++++ pwd
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''h60c717f3'
053-pip-ctrlin           - 1m50s: ++ export SEEDN=18
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: ++ SEEDN=18
053-pip-ctrlin           - 1m50s: +++ echo build/1/specimen_020
053-pip-ctrlin           - 1m50s: +++ md5sum
053-pip-ctrlin           - 1m50s: ++ echo build/1/specimen_018
053-pip-ctrlin           - 1m50s: ++ echo build/1/specimen_017
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: ++ md5sum
053-pip-ctrlin           - 1m50s: ++ md5sum
053-pip-ctrlin           - 1m50s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin/build/1/specimen_019
053-pip-ctrlin           - 1m50s: ++ cut -c1-8
053-pip-ctrlin           - 1m50s: ++ cut -c1-8
053-pip-ctrlin           - 1m50s: +++ cut -c1-8
053-pip-ctrlin           - 1m50s: ++ export SEEDN=19
053-pip-ctrlin           - 1m50s: ++ SEEDN=19
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: ++ echo build/1/specimen_019
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''h9aa8f41f'
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: ++ md5sum
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''hc045d7c3'
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: ++ cut -c1-8
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: ++ export SEED=363f37c0
053-pip-ctrlin           - 1m50s: ++ SEED=363f37c0
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''h9fd99230'
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: +++ sed 's/specimen_0*//'
053-pip-ctrlin           - 1m50s: ++++ pwd
053-pip-ctrlin           - 1m50s: +++ basename /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/fuzzers/053-pip-ctrlin/build/1/specimen_020
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: ++ export SEEDN=20
053-pip-ctrlin           - 1m50s: ++ SEEDN=20
053-pip-ctrlin           - 1m50s: ++ echo build/1/specimen_020
053-pip-ctrlin           - 1m50s: ++ md5sum
053-pip-ctrlin           - 1m50s: ++ cut -c1-8
053-pip-ctrlin           - 1m50s: + echo '`define SEED 32'\''h363f37c0'
053-pip-ctrlin           - 1m50s: + make -f /tmpfs/src/github/symbiflow-prjxray-continuous-db-artix7/utils/top_generate.mk
053-pip-ctrlin           - 1m50s: make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
053-pip-ctrlin           - 1m50s: make[3]: *** [vivado.ok] Error 1
053-pip-ctrlin           - 1m50s: make[2]: *** [build/1/specimen_008/OK] Error 2
053-pip-ctrlin           - 1m50s: make[2]: *** Waiting for unfinished jobs....
053-pip-ctrlin           - 1m50s: make[3]: *** [vivado.ok] Error 1
053-pip-ctrlin           - 1m50s: make[2]: *** [build/1/specimen_019/OK] Error 2
053-pip-ctrlin           - 1m50s: make[1]: *** [run] Error 2
053-pip-ctrlin           - 1m50s: 
053-pip-ctrlin           - 1m50s: --------------------------------------------------------------------------
053-pip-ctrlin           - 1m50s: Failed @ 2019-02-05T14:12:38.089913 with exit code: 2
Makefile:75: recipe for target '053-pip-ctrlin/run.ok' failed
make: *** [053-pip-ctrlin/run.ok] Error 2
make: *** Waiting for unfinished jobs....
```

--- a/fuzzers/053-pip-ctrlin/generate.tcl
+++ b/fuzzers/053-pip-ctrlin/generate.tcl
@@ -29,8 +29,11 @@ for {gets $fp line} {$line != ""} {gets $fp line} {
 }
 close $fp
 
-set int_l_tiles [randsample_list [llength $todo_lines] [filter [pblock_tiles roi] {TYPE == INT_L}]]
-set int_r_tiles [randsample_list [llength $todo_lines] [filter [pblock_tiles roi] {TYPE == INT_R}]]
+# each run can fail up to three times so we need to prepare 3*todo_lines tiles to work on
+set tiles [expr 3 * [llength $todo_lines]]
+
+set int_l_tiles [randsample_list $tiles [filter [pblock_tiles roi] {TYPE == INT_L}]]
+set int_r_tiles [randsample_list $tiles [filter [pblock_tiles roi] {TYPE == INT_R}]]
 
 for {set idx 0} {$idx < [llength $todo_lines]} {incr idx} {
     set line [lindex $todo_lines $idx]
@@ -39,26 +42,54 @@ for {set idx 0} {$idx < [llength $todo_lines]} {incr idx} {
     set tile_type [lindex $line 0]
     set dst_wire [lindex $line 1]
     set src_wire [lindex $line 2]
-
-    if {$tile_type == "INT_L"} {set tile [lindex $int_l_tiles $idx]; set other_tile [lindex $int_r_tiles $idx]}
-    if {$tile_type == "INT_R"} {set tile [lindex $int_r_tiles $idx]; set other_tile [lindex $int_l_tiles $idx]}
-
-    set driver_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
-            -of_objects [get_nodes -of_objects [get_wires $other_tile/CLK*0]]]]]
-
-    set recv_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
-            -of_objects [get_nodes -of_objects [get_wires $tile/$dst_wire]]]]]
-
     set mylut [create_cell -reference LUT1 mylut_$idx]
-    set_property -dict "LOC $driver_site BEL A6LUT" $mylut
-
     set myff [create_cell -reference FDRE myff_$idx]
-    set ffbel [lindex "AFF A5FF BFF B5FF CFF C5FF DFF D5FF" [expr {int(rand()*8)}]]
-    set_property -dict "LOC $recv_site BEL $ffbel" $myff
-
     set mynet [create_net mynet_$idx]
-    connect_net -net $mynet -objects "$mylut/O $myff/R"
-    route_via $mynet "$tile/$src_wire $tile/$dst_wire"
+
+    set tries 0
+    while {1} {
+        set tile_idx [expr $tries + [expr $idx * 3]]
+        incr tries
+
+        if {$tile_type == "INT_L"} {set tile [lindex $int_l_tiles $tile_idx]; set other_tile [lindex $int_r_tiles $tile_idx]}
+        if {$tile_type == "INT_R"} {set tile [lindex $int_r_tiles $tile_idx]; set other_tile [lindex $int_l_tiles $tile_idx]}
+
+        set driver_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
+                -of_objects [get_nodes -of_objects [get_wires $other_tile/CLK*0]]]]]
+
+        set recv_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
+                -of_objects [get_nodes -of_objects [get_wires $tile/$dst_wire]]]]]
+
+        set_property -dict "LOC $driver_site BEL A6LUT" $mylut
+
+        set ffbel [lindex "AFF A5FF BFF B5FF CFF C5FF DFF D5FF" [expr {int(rand()*8)}]]
+        set_property -dict "LOC $recv_site BEL $ffbel" $myff
+
+        connect_net -net $mynet -objects "$mylut/O $myff/R"
+        set rc [route_via $mynet "$tile/$src_wire $tile/$dst_wire" 0]
+        if {$rc != 0} {
+            puts "ROUTING DONE!"
+            break
+        }
+
+        # fallback
+        puts "WARNING: failed to route net"
+        write_checkpoint -force route_todo_$idx.$tries.fail.dcp
+
+        puts "Rolling back route"
+        set_property is_route_fixed 0 $mynet
+        set_property is_bel_fixed 0 $mylut
+        set_property is_loc_fixed 1 $mylut
+        set_property is_bel_fixed 0 $myff
+        set_property is_loc_fixed 1 $myff
+        route_design -unroute -nets $mynet
+
+        # sometimes it gets stuck in specific src -> dst locations
+        if {$tries >= 3} {
+            puts "WARNING: failed to route net after $tries tries"
+            error
+        }
+    }
 }
 
 route_design

--- a/fuzzers/055-pip-gnd/generate.tcl
+++ b/fuzzers/055-pip-gnd/generate.tcl
@@ -28,8 +28,12 @@ for {gets $fp line} {$line != ""} {gets $fp line} {
 }
 close $fp
 
-set int_l_tiles [randsample_list [llength $todo_lines] [filter [pblock_tiles roi] {TYPE == INT_L}]]
-set int_r_tiles [randsample_list [llength $todo_lines] [filter [pblock_tiles roi] {TYPE == INT_R}]]
+# each run can fail up to three times so we need to prepare 3*todo_lines tiles to work on
+set tiles [expr 3 * [llength $todo_lines]]
+
+set int_l_tiles [randsample_list $tiles [filter [pblock_tiles roi] {TYPE == INT_L}]]
+set int_r_tiles [randsample_list $tiles [filter [pblock_tiles roi] {TYPE == INT_R}]]
+
 
 for {set idx 0} {$idx < [llength $todo_lines]} {incr idx} {
     set line [lindex $todo_lines $idx]
@@ -39,15 +43,8 @@ for {set idx 0} {$idx < [llength $todo_lines]} {incr idx} {
     set dst_wire [lindex $line 1]
     set src_wire [lindex $line 2]
 
-    if {$tile_type == "INT_L"} {set tile [lindex $int_l_tiles $idx]; set other_tile [lindex $int_r_tiles $idx]}
-    if {$tile_type == "INT_R"} {set tile [lindex $int_r_tiles $idx]; set other_tile [lindex $int_l_tiles $idx]}
-
-    set recv_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
-            -of_objects [get_nodes -of_objects [get_wires $other_tile/CLK*0]]]]]
-
     set mynet [create_net mynet_$idx]
     set mylut [create_cell -reference LUT1 mylut_$idx]
-    set_property -dict "LOC $recv_site BEL A6LUT" $mylut
 
     if {$src_wire == "VCC_WIRE"} {
         set mytie [create_cell -reference VCC mytie_$idx]
@@ -59,7 +56,41 @@ for {set idx 0} {$idx < [llength $todo_lines]} {incr idx} {
         connect_net -net $mynet -objects "$mytie/G $mylut/I0"
     }
 
-    route_via $mynet "$tile/$src_wire $tile/$dst_wire"
+    set tries 0
+    while {1} {
+        set tile_idx [expr $tries + [expr $idx * 3]]
+        incr tries
+
+        if {$tile_type == "INT_L"} {set tile [lindex $int_l_tiles $tile_idx]; set other_tile [lindex $int_r_tiles $tile_idx]}
+        if {$tile_type == "INT_R"} {set tile [lindex $int_r_tiles $tile_idx]; set other_tile [lindex $int_l_tiles $tile_idx]}
+
+        set recv_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
+                -of_objects [get_nodes -of_objects [get_wires $other_tile/CLK*0]]]]]
+
+        set_property -dict "LOC $recv_site BEL A6LUT" $mylut
+
+        set rc [route_via $mynet "$tile/$src_wire $tile/$dst_wire" 0]
+        if {$rc != 0} {
+            puts "ROUTING DONE!"
+            break
+        }
+
+        # fallback
+        puts "WARNING: failed to route net"
+        write_checkpoint -force route_todo_$idx.$tries.fail.dcp
+
+        puts "Rolling back route"
+        set_property is_route_fixed 0 $mynet
+        set_property is_bel_fixed 0 $mylut
+        set_property is_loc_fixed 1 $mylut
+        route_design -unroute -nets $mynet
+
+        # sometimes it gets stuck in specific src -> dst locations
+        if {$tries >= 3} {
+            puts "WARNING: failed to route net after $tries tries"
+            error
+        }
+    }
 }
 
 route_design

